### PR TITLE
codeintel: Add diagnostics route to bundle manager

### PR DESCRIPTION
**Pull request by efritz**

### CODE REVIEW

File: cmd/precise-code-intel-bundle-manager/internal/database/database.go [hunk 1]
```diff
@@ -192,6 +195,41 @@ func (db *databaseImpl) Hover(ctx context.Context, path string, line, character
...
+// Diagnostics returns the diagnostics for the documents that have the given path prefix.
+func (db *databaseImpl) Diagnostics(ctx context.Context, prefix string) ([]client.Diagnostic, error) {
+	paths, err := db.getPathsWithPrefix(ctx, prefix)
+	if err != nil {
+		return nil, pkgerrors.Wrap(err, "db.getPathsWithPrefix")
+	}
+
+	var diagnostics []client.Diagnostic
+	for _, path := range paths {
+		documentData, exists, err := db.getDocumentData(ctx, prefix)
```
Comment (gbrik) [inline cmd/precise-code-intel-bundle-manager/internal/database/database.go:207]: typo:
`db.getDocumentData(ctx, path)`?

File: cmd/precise-code-intel-bundle-manager/internal/database/database.go [hunk 2]
```diff
@@ -192,6 +195,41 @@ func (db *databaseImpl) Hover(ctx context.Context, path string, line, character
...
 
+// Diagnostics returns the diagnostics for the documents that have the given path prefix.
+func (db *databaseImpl) Diagnostics(ctx context.Context, prefix string) ([]client.Diagnostic, error) {
+	paths, err := db.getPathsWithPrefix(ctx, prefix)
+	if err != nil {
+		return nil, pkgerrors.Wrap(err, "db.getPathsWithPrefix")
+	}
+
+	var diagnostics []client.Diagnostic
+	for _, path := range paths {
```
Comment (gbrik) [inline cmd/precise-code-intel-bundle-manager/internal/database/database.go:206]: nit:
would it be hard to batch this?

File: internal/codeintel/bundles/client/types.go [hunk 3]
```diff
@@ -32,3 +32,17 @@ type PackageInformationData struct {
...
+	Path           string `json:"path"`
+	Severity       int    `json:"severity"`
+	Code           string `json:"code"`
+	Message        string `json:"message"`
+	Source         string `json:"source"`
+	StartLine      int    `json:"startLine"`
+	StartCharacter int    `json:"startCharacter"`
+	EndLine        int    `json:"endLine"`
+	EndCharacter   int    `json:"endCharacter"`
+}
```
Comment (gbrik) [inline internal/codeintel/bundles/client/types.go:48]: deja vu

File: cmd/precise-code-intel-bundle-manager/internal/database/database.go (hunk 2 above)
Comment (efritz) [reply cmd/precise-code-intel-bundle-manager/internal/database/database.go:206]: No, but I think the dominating factor here is the deserialization and not the underlying query. And we'll need to add paging/limits to this, so batching now would make that logic more complicated.

**Pull request merged**